src/seed/service.ml

```diff
@@ -0,0 +1,66 @@
+open Base
+module Sig = Sig
+
+let registered_seeds = ref (Map.empty (module String))
```

Comment:
Why are you using mutable state?

Reply:
Because I want to be able to register seeds after the service was started. I don't see how to avoid mutable state here.